frodo.js

```diff
@@ -5,118 +5,130 @@
  * for the extras/ plugins if they are loaded into a presentation.
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.